### ISSUE TITLE
UTF-8 to UTF7-IMAP might break the password

### ISSUE THREAD

**Issue by commanddotcom** [barbushin/php-imap#242]
In method `imap()` we convert `$args` to UTF7-IMAP. But here is the trick: 
```php
$password = "my_password&!";
// imap_utf7_encode() is original function
echo imap_utf7_encode($password); // return "my_password&-!" 
// mb_convert_encoding() is just better but it doesn't help here
echo mb_convert_encoding($password, "UTF7-IMAP", "UTF-8"); // return "my_password&-!"
```
In this case authorization will fail because the password was changed. So we suppose to have exceptions here.

**Comment by commanddotcom** [comment]
`imap_reopen` method require exception as well 
and `imap_search`...

**Comment by commanddotcom** [comment]
@barbushin this is definitely a bug

**Comment by ComradePashka** [comment]
I got same evil bug and spent few hours to check my environment, settings, configs. I tried to use imap_ calls directly at VPS and saw that everything works just fine.
So finally I have added some debug messages near https://github.com/barbushin/php-imap/blob/d1aee641fbf12aeba6a198657bad98ce29e0f665/src/PhpImap/Mailbox.php#L856 and finally got it! -- exact same behaviour with password containing `&` character.
Password passed in $args array definitely must NOT be encoded by `imap_utf7_encode()`
